File: src/main/java/io/nightfis/createstellarexoduscore/filter/HasCustomNameFilter.java
package io.nightfis.createstellarexoduscore.filter;

import net.minecraft.network.chat.Component;
import net.minecraft.world.entity.LivingEntity;

public class HasCustomNameFilter extends TargetFilter {

    private static final String ID = "HasCustomNameFilter";

    @Override
    public String getId() {
        return ID;
    }

    @Override
    public boolean match(LivingEntity entity, String arg) {
        return entity.getCustomName() != null;
    }

    @Override
    public Component title() {
        return Component.translatable("screen.createstellarexoduscore.auto_aim_turret.filter.has_custom_name");
    }
}

File: src/main/kotlin/io/nightfis/createstellarexoduscore/filter/HasCustomNameFilter.kt
package io.nightfis.createstellarexoduscore.filter

import net.minecraft.network.chat.Component
import net.minecraft.world.entity.LivingEntity

class HasCustomNameFilter : TargetFilter() {

    override fun getId(): String = ID

    override fun match(entity: LivingEntity, arg: String): Boolean {
        return entity.customName != null
    }

    override fun title(): Component {
        return Component.translatable("screen.create_stellar_exodus_core.auto_aim_turret.filter.has_custom_name")
    }

    companion object {
        private const val ID = "HasCustomNameFilter"
    }
}
